[ServicesStage] Index out of range when checking task list

Hi team,

I got a panic when I deploy a component.

```
PANIC: runtime error: index out of range [22] with length 20
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: 
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: goroutine 77 [running]:
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/servicestage.componentInstanceRefreshFunc.func1()
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: 	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/servicestage/resource_huaweicloud_servicestage_component_instance.go:1382 +0xcf
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/resource/state.go:110 +0x207
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
2022-07-05T09:13:06.432+0800 [DEBUG] provider.terraform-provider-huaweicloud_v1.37.1.exe: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/resource/state.go:83 +0x1d8
2022-07-05T09:13:06.433+0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-07-05T09:13:06.433+0800 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = transport is closing"
2022-07-05T09:13:06.443+0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/local-registry/huaweicloud/huaweicloud/1.37.1/windows_amd64/terraform-provider-huaweicloud_v1.37.1.exe pid=29964 error="exit status 2"
```

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.14
 
 require (
 	github.com/GehirnInc/crypt v0.0.0-20200316065508-bb7000b8a962
-	github.com/chnsz/golangsdk v0.0.0-20220701061130-f5ff51d74f49
+	github.com/chnsz/golangsdk v0.0.0-20220705081403-19cda7d0379b
 	github.com/hashicorp/errwrap v1.1.0 // indirect
 	github.com/hashicorp/go-cleanhttp v0.5.2
 	github.com/hashicorp/go-multierror v1.1.1

--- a/huaweicloud/services/servicestage/resource_huaweicloud_servicestage_component_instance.go
+++ b/huaweicloud/services/servicestage/resource_huaweicloud_servicestage_component_instance.go
@@ -1446,16 +1446,20 @@ func resourceComponentInstanceDelete(ctx context.Context, d *schema.ResourceData
 	return nil
 }
 
-func componentInstanceRefreshFunc(c *golangsdk.ServiceClient, instance string) resource.StateRefreshFunc {
+func componentInstanceRefreshFunc(c *golangsdk.ServiceClient, jobId string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		resp, err := jobs.Get(c, instance)
+		opt := jobs.ListOpts{
+			Limit: 50,
+		}
+		resp, err := jobs.List(c, jobId, opt)
 		if err != nil {
 			return resp, "ERROR", err
 		}
-		if resp.TaskCount < 1 {
+		rl := len(resp)
+		if rl < 1 {
 			return resp, "NO TASK", nil
 		}
-		return resp, resp.Tasks[resp.TaskCount-1].Status, nil
+		return resp, resp[rl-1].Status, nil
 	}
 }
 

